internal/v2/remote_signer/remote_signer_test.go

```diff
@@ -1,28 +1,173 @@
+	log.Printf("Client: connected to: %s", address)
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTimeout)
+	defer cancel()
+
+	// Setup bidrectional streaming session.
+	callOpts := []grpc.CallOption{}
+	cstream, err := c.SetUpSession(ctx, callOpts...)
+	if err != nil  {
+		log.Fatalf("Client: failed to setup bidirectional streaming RPC session: %v", err)
+	}
+	log.Printf("Client: set up bidirectional streaming RPC session.")
+
+	// TODO(rmehta19): convert into a for loop when adding more tests.
+	// Setup data for testing Sign
+	clientTlsCert, err := tls.X509KeyPair(clientCertPEM, clientKeyPEM)
+	if err != nil {
+		log.Fatalf("tls.X509KeyPair failed: %v", err)
+	}
+	clientx509Cert, err := x509.ParseCertificate(clientCertDER)
+	if err != nil {
+		log.Fatalf("failed to parse cert: %v", err)
+	}
+	s := New(clientx509Cert, cstream, "client_hostname")
+	testInBytes := []byte("Test data.")
+	
+	// TODO(rmehta19): Investigate whether go crypto libraries compute hash.
+	// If so, remove this line, and just pass testInBytes as digest.
+	hsha256 := sha256.Sum256([]byte(testInBytes))
+	signerOpts := crypto.SHA256
+
+	// Test Sign.
+	gotSignedBytes, err := s.Sign(rand.Reader, hsha256[:], signerOpts)
+	if err != nil {
+		t.Errorf("call to remote signer Sign API failed: %v", err)
+	}
+	wantSignedBytes, err := clientTlsCert.PrivateKey.(crypto.Signer).Sign(rand.Reader, hsha256[:], signerOpts)
+	if err != nil {
+		t.Errorf("call to Sign API failed: %v", err)
+	}
+	if !bytes.Equal(gotSignedBytes, wantSignedBytes) {
+		t.Errorf("gotSignedBytes = %v, wantSignedBytes = %v", gotSignedBytes, wantSignedBytes)
+	}
+	stop()
+}
+
+// TestNew runs unit test for New.
+func TestNew(t *testing.T) {
+	// Start up fake S2Av2 server.
+	var wg sync.WaitGroup
+	wg.Add(1)
+	stop, address, err := startFakeS2Av2Server(&wg)
+	wg.Wait()
+	if err != nil {
+		log.Fatalf("error starting fake S2Av2 Server: %v", err)
+	}
+
+	// Create stream to S2Av2.
+	opts := []grpc.DialOption {
+		grpc.WithTransportCredentials(insecure.NewCredentials()),
+		grpc.WithReturnConnectionError(),
+		grpc.WithBlock(),
+	}
+	conn, err := grpc.Dial(address, opts...)
+	if err != nil {
+		log.Fatalf("Client: failed to connect: %v", err)
+	}
+	defer conn.Close()
+	c := s2av2pb.NewS2AServiceClient(conn)
+	log.Printf("Client: connected to: %s", address)
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTimeout)
+	defer cancel()
+
+	// Setup bidrectional streaming session.
+	callOpts := []grpc.CallOption{}
+	cstream, err := c.SetUpSession(ctx, callOpts...)
+	if err != nil  {
+		log.Fatalf("Client: failed to setup bidirectional streaming RPC session: %v", err)
+	}
+	log.Printf("Client: set up bidirectional streaming RPC session.")
+
+	// Setup data for testing New.
+	clientx509Cert, err := x509.ParseCertificate(clientCertDER)
+	if err != nil {
+		t.Errorf("failed to parse cert: %v", err)
+	}
+
+	// Test New.
+	got := New(clientx509Cert, cstream, "client_hostname")
```

Comment:
We're not really using that this is a real stream - should we just pass in a fake stream instead?